tests/foreman/virtwho/utils.py

```diff
@@ -0,0 +1,185 @@
+"""Utility module to handle the virtwho configure UI/CLI/API testing"""
+import re
+import time
+from robottelo import ssh
+from robottelo.config import settings
+
+
+class VirtWhoError(Exception):
+    """Exception raised for failed virtwho operations"""
+
+
+def get_system(system_type):
+    """Return a dict account for ssh connect.
+    :param str system_type: The type of the system, should be one of
+        ('satellite', 'guest').
+    :raises: VirtWhoError: If wrong ``system_type`` specified.
+    """
+    if system_type == 'guest':
+        return{
+            'hostname': settings.virtwho.guest,
+            'username': settings.virtwho.guest_username,
+            'password': settings.virtwho.guest_password,
+        }
+    elif system_type == 'satellite':
+        return{
+            'hostname': settings.server.hostname,
+            'username': settings.server.ssh_username,
+            'password': settings.server.ssh_password,
+        }
+    else:
+        raise VirtWhoError(
+            '"{}" system type is not supported. Please use one of {}'
+            .format(system_type, ('satellite', 'guest'))
+        )
+
+
+def runcmd(cmd, system=None, timeout=None, port=22):
+    """Return the retcode and stdout.
+    :param str cmd: The command line will be executed in the target system.
+    :param dict system: the system account which ssh will connect to,
+        it will connect to the satellite host if the system is None.
+    :param int timeout: Time to wait for establish the connection.
+    :param int port: The server port to ssh connect tol
+    """
+    system = system or get_system('satellite')
+    hostname = system['hostname']
+    username = system['username']
+    password = system['password']
+    if ":" in hostname:
+        hostname, port = hostname.split(':')
```

Comment:
For the kubevirt vm, we have to connect it with a different port, in virt-who ci framework, we often define the guest such as: `guest=10.xxx.xxx.xxx:32302`.
But I think we should a new option in the robottelo.properties for the guest port, such as:
```
guest=
guest_port=
```

Reply:
Meanwhile,  I still suggest to keep `runcmd()` function for virtwho, the reason is:
1. We have to do many 'Text Retrieval' for the `result = ssh.command`, but you know `result.stdout` is a long list(). It will take more time if foreach line to get the value from rhsm.log .

2.  I'm not sure is it more pythonicI if using `ssh.command(cmd, **system)` so frequently in other functions, take register_system() as an example, if without runcmd(), it will look like:
```
def register_system(system, activation_key=None, org='Default_Organization'):
    ssh.command('subscription-manager unregister', **system)
    ssh.command('subscription-manager clean', **system)
    ssh.command('rpm -qa | grep katello-ca-consumer | xargs rpm -e |sort', **system)
    cmd = 'subscription-manager register --org={} '.format(org)
    result =  ssh.command(cmd, **system)
    if result.retcode == 0 or "system has been registered" in '\n'.join(result.stdout):
        return True
    .........
```

Reply:
1. Adding `guest_port` option makes sense to me.
2. > It will take more time if foreach line to get the value from rhsm.log .

    I don't understand what you mean. It doesn't really matter if you have thin function that obtains list of strings, joins them into single big string and returns that, or if you get list of strings and make it responsibility of caller to join that list on their own end. Establishing connection and transferring content of file will be slowest thing in both cases.

3. Personally, I don't see much difference between these two:
```
ssh.command('subscription-manager unregister', **system)
```
```
runcmd('subscription-manager unregister', system)
```

As I said before, I don't find that blocking, so if you want to have thin wrapper around our `ssh.command`, I'm OK with that.